src/Engines/thinEngine.ts

```diff
@@ -453,7 +456,7 @@ export class ThinEngine {
     /** @hidden */
     protected _boundTexturesCache: { [key: string]: Nullable<InternalTexture> } = {};
     /** @hidden */
-    protected _currentEffect: Nullable<Effect>;
+    public _currentEffect: Nullable<Effect>;

```

Comment:
Is there a way to avoid the public ?

Reply:
I reverted it back to `protected` as I checked the value of `_currentEffect` in `engine.externalTexture` but I really didn't have to do it.